lib/fedex/tracking_information.rb

```diff
@@ -47,11 +47,15 @@ def initialize(details = {})
       if details.has_key?(:estimated_delivery_timestamp)
         @estimated_delivery_at = Time.parse(details[:estimated_delivery_timestamp])
       end
+      
+      if details[:events].select { |event| event[:event_type] == "PU" }.first
```

Comment:
Also:

d) are we guaranteed that the events will be in chronological order?

e) should we check for "PX" events as well?  (https://github.com/stitchfix/spectre/blob/8112f6c46454deef6a30ce4e179741921ab60032/app/services/fedex_status_service.rb#L65)  (I don't know the difference, I just know that both exist).